packages-exp/auth-exp/src/core/mfa/mfa_user.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as externs from '@firebase/auth-types-exp';
+
+import { User } from '../../model/user';
+import { MultiFactorSessionType, MultiFactorSession } from './mfa_session';
+import { MultiFactorAssertion } from './assertions';
+import { withdrawMfa } from '../../api/account_management/mfa';
+import { AuthErrorCode } from '../errors';
+
+export class MultiFactorUser implements externs.MultiFactorUser {
+  // TODO(avolkovi): Set these correctly from getAccountInfo on reload
+  enrolledFactors: externs.MultiFactorInfo[] = [];
+
+  constructor(readonly user: User) {}
+
+  async getSession(): Promise<MultiFactorSession> {
```

Comment:
Is there an externs type for `MultiFactorSession` that we should use here instead?

Reply:
I had a question about this actually, should we be using externs on the return types?

Reply:
done